### PR TITLE
Support GitHub Flavored Markdown

### LINKED ISSUE #28: Relative remote images do not render
Currently, the following happens:

- Either the URL is absolute and the image is retrieved over the network.
- The URL is deemed invalid/ an imageDirectory is set, but the image is always loaded as if it was locally.

An Dart IO Directory can also point to the network. Therefore, wouldn't it be better to test if the imageDirectory + path equals to a network path before retrieving the image?

### LINKED ISSUE #29: Add support for tables
As https://github.com/flutter/flutter/issues/8240 asks, please add support for Markdown tables.

### LINKED ISSUE #31: more features
I noticed flutter_markdown is lacking support:
- space between paragraph.
for beauty text since markdown is the best way to implement complex text.
currently only support code(pre tag)

- custom Image(Widget) creating.
cache image, add customized ImageView

Need these features? I can send a PR for these.

### LINKED ISSUE #40: Add support for line breaks
Maybe I'm missing something, but I don't see a way to add line breaks where wanted.  A lot of Markdown supports this with the following rule: "To force a line return, place two empty spaces at the end of a line."  

I can get a maximum of one new line, after that, they seem to collapse.  I'd like to have something like
```
Hello
  
  
  
There are three lines of space above this
```

### LINKED ISSUE #77: Support user defined parse funcs
i read the source code, and i found that **markdown's extensionSet**  methods give access to parse md string, why not export out this attribute to user?

### LINKED ISSUE #113: is there plan to support other flavored markdown?
like the [github flavorer](https://github.github.com/gfm/)?

or a plugin system and let us create our own syntax?

### LINKED ISSUE #124: overflowed line of code in <pre> should be scrollable, instead of wrapped
A line of code may be super long, and the common practice is to make the \<pre\> horizontal scrollable, instead of wrapping the code.

An example:

Screen Shot 2019-09-19 at 13 44 40

### LINKED ISSUE #126: incorrect bottom padding in <pre>
# repro case

~~~
```
line 1
line 2
line 3
line 4
```
~~~

# actually

Screen Shot 2019-09-19 at 14 31 14

# expected

proper bottom padding

# research

Not sure if this is about a suffix \n or wrong bottom padding

### LINKED ISSUE #130: MarkdownBody.shrinkwrap
Currently, MarkdownBody expands to fill all available space, but that is often not ideal. For example, I wanted to use it to display some dynamic content in a bottomSheet, but that forces the sheet to expand to the full size of the screen.

I was able to implement a solution by adding a `shrinkWrap` property, and then setting `mainAxisSize` on the `Column` it creates.

```
// MarkdownBody
  Widget build(BuildContext context, List<Widget> children) {
    if (children.length == 1 && !shrinkWrap) return children.single;
    return new Column(
      mainAxisSize: shrinkWrap ? MainAxisSize.min : MainAxisSize.max,
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: children,
    );
  }
```

This seems to work well. I don't have the ability now to do robust testing on other scenarios, but since this is gated behind a new property, it shouldn't break any existing content.

Happy to submit this as a pull-request if this project is still active.

### PR DESCRIPTION
**Pull request by KagurazakaHanabi**
## Related Issues
- [x] Closes #28 
- [x] Closes #29 
- [x] Closes #31 
- [x] Closes #40 
- [x] Closes #77 
- [x] Closes #113 
- [x] Closes #124 
- [x] Closes #126 
- [x] Closes #130

### CODE REVIEW

**Comment by KagurazakaHanabi** [comment]
Screenshot_2019-10-19-22-49-28

File: lib/src/widget.dart [hunk 1]
```diff
@@ -154,12 +178,12 @@ class _MarkdownWidgetState extends State<MarkdownWidget>
 /// A non-scrolling widget that parses and displays Markdown.
 ///
 /// Supports all standard Markdown from the original
```
Comment (domesticmouse) [inline lib/src/widget.dart]: This should probably be changed to 'Github flavoured markdown', not Markdown from the original.

File: lib/src/widget.dart [hunk 2]
```diff
@@ -191,12 +226,12 @@ class MarkdownBody extends MarkdownWidget {
 /// A scrolling widget that parses and displays Markdown.
 ///
 /// Supports all standard Markdown from the original
```
Comment (domesticmouse) [inline lib/src/widget.dart]: Ditto

File: test/flutter_markdown_test.dart [hunk 3]
```diff
@@ -44,6 +44,15 @@ void main() {
     _expectTextStrings(widgets, <String>['strikethrough']);
   });
 
+  testWidgets('Line break', (WidgetTester tester) async {
```
Comment (domesticmouse) [inline test/flutter_markdown_test.dart]: Can you please add tests for the new Github flavoured markdown that you have added.

Comment (KagurazakaHanabi) [reply test/flutter_markdown_test.dart]: Done

**Comment by ThinkDigitalSoftware** [comment]
We're missing an update to changelog.md

**Pull request merged**